sa/sa.go

```diff
@@ -1622,31 +1614,14 @@ func (ssa *SQLStorageAuthority) getAuthorizations(ctx context.Context, table str
 			continue
 		}
 
-		// Retrieve challenges for the authzvar challObjs []challModel
-		var challObjs []challModel
-		_, err = ssa.dbMap.Select(
-			&challObjs,
-			getChallengesQuery,
-			map[string]interface{}{"authID": auth.ID},
-		)
-		if err != nil {
-			return nil, err
-		}
-		var challs []core.Challenge
-		for _, c := range challObjs {
-			chall, err := modelToChallenge(&c)
-			if err != nil {
-				return nil, err
-			}
-			challs = append(challs, chall)
-		}
-		auth.Challenges = challs
-
 		if auth.Identifier.Type != core.IdentifierDNS {
 			return nil, fmt.Errorf("unknown identifier type: %q on authz id %q", auth.Identifier.Type, auth.ID)
 		}
 		existing, present := byName[auth.Identifier.Value]
 		if !present || auth.Expires.After(*existing.Expires) {
+			// Retrieve challenges for the authz
+			auth.Challenges, err = ssa.getChallenges(auth.ID)
```

Comment:
I think this new call to `ssa.getChallenges` should be captured by the new feature flag in case it causes problems. I'm slightly worried the increase in queries might be problematic.

If the SA looks at the feature flag we also need to update the next config for the SA to add it.